### PR TITLE
#24 Add user card

### LINKED ISSUE #24: Make user card component which show main information about user
**frontend part**

- [x]  add the user to my friends
- [x]  show the button "add friend" if the user not in my friend list
- [x] don't show the button "add friend" if the user in my friend list
- [x] show the user status online or not
- [x] route to chat page after click on button "message"
- [x] show information about the user(profile picture,firstName, lastName, languages which user speaks and want to learn and their level)
#24

### PR DESCRIPTION
**Pull request by OrestHaponov**

### CODE REVIEW

File: frontend/src/components/pages/profile/profile.scss [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.container {
+  width: 944px;
```
Comment (apastuhov) [inline frontend/src/components/pages/profile/profile.scss]: remove after integration with Layout

File: frontend/src/components/shared/userCard/UserCard.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import React, { MouseEvent } from 'react';
+import Box from '@material-ui/core/Box';
+import { Link } from 'react-router-dom';
+import UserPhoto from '../../util/user.png';
+import './userCard.scss';
+import User from '../../util/user.json';
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/UserCard.tsx]: user-mock.json

File: frontend/src/components/shared/userCard/UserCard.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  let { firstName, lastName, age, country, speaks, learn, online, isFriend } = User;
+
+  const sendFriendRequest = (event: MouseEvent) => {
+    event.preventDefault();
+  };
+
+  return (
+    <Box boxShadow={2} className="user-card">
+      <div className="leftbar">
+        <img src={UserPhoto} alt="" />
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/UserCard.tsx]: fullfill alt like "photo of {username}"

File: frontend/src/components/shared/userCard/UserCard.tsx [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+import { Link } from 'react-router-dom';
+import UserPhoto from '../../util/user.png';
+import './userCard.scss';
+import User from '../../util/user.json';
+
+export const UserCard: React.FC = () => {
+  let { firstName, lastName, age, country, speaks, learn, online, isFriend } = User;
+
+  const sendFriendRequest = (event: MouseEvent) => {
+    event.preventDefault();
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/UserCard.tsx:14]: console.log("Add XYZ to friend");

File: frontend/src/components/shared/userCard/UserCard.tsx (hunk 3 above)
Comment (apastuhov) [inline frontend/src/components/shared/userCard/UserCard.tsx]: move image to separate component

File: frontend/src/components/shared/userCard/userCard.scss [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  border-radius: $block-round;
+  .leftbar {
+    position: relative;
+    width: 180px;
+    text-align: center;
+    border-right: 2px solid $secondary-color;
+    .online-status {
+      display: inline-block;
+      position: absolute;
+      top: 95px;
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/userCard.scss]: use bottom

File: frontend/src/components/shared/userCard/userCard.scss [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+      right: 35px;
+      width: 20px;
+      height: 20px;
+      border: 2px solid $white;
+      border-radius: 15px;
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/userCard.scss]: 50%

File: frontend/src/components/shared/userCard/userCard.scss [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    }
+    h3 {
+      @include title;
+      margin: $small-retreat;
+    }
+    p {
+      @include text;
+      margin: $small-retreat;
+      color: $gray;
+      span {
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/userCard.scss]: .flag

File: frontend/src/components/shared/userCard/userCard.scss [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+    }
+    .speaks,
+    .learning {
+      h3 {
+        color: $gray;
+        margin: 0;
+      }
+      .languages {
+        display: flex;
+        padding-left: 13px;
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/userCard.scss]: variable 14

File: frontend/src/components/shared/userCard/userCard.scss [hunk 9]
```diff
@@ -0,0 +1,122 @@
...
+      margin-bottom: 55px;
+    }
+    .speaks,
+    .learning {
+      h3 {
+        color: $gray;
+        margin: 0;
+      }
+      .languages {
+        display: flex;
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/userCard.scss:77]: review css-grid usage

File: frontend/src/components/shared/userCard/userCard.scss [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+        }
+        p {
+          @include text;
+          padding-left: $small-retreat;
+          margin: $small-retreat 0 0 0;
+        }
+      }
+    }
+    .buttons-action {
+      position: absolute;
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/userCard.scss]: use flex

File: frontend/src/components/shared/userCard/userCard.scss [hunk 11]
```diff
@@ -0,0 +1,122 @@
...
+      .languages {
+        display: flex;
+        padding-left: 13px;
+        margin-bottom: $small-retreat;
+        p:first-child {
+          min-width: 95px;
+        }
+        p {
+          @include text;
+          padding-left: $small-retreat;
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/userCard.scss]: gap-4
gap-8
gap-12
gap-16
gap-24
gap-36

File: frontend/src/components/shared/userCard/userCard.scss [hunk 12]
```diff
@@ -0,0 +1,122 @@
...
+        background: $primary-color;
+      }
+      .send-message {
+        background: $secondary-color;
+      }
+    }
+  }
+}
+
+@media screen and (max-width: 540px) {
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/userCard.scss]: to variable

File: frontend/src/components/shared/userCard/userCard.scss [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+  .user-card {
+    flex-direction: column;
```
Comment (apastuhov) [inline frontend/src/components/shared/userCard/userCard.scss]: to selector as \@ media mobile

**Review by apastuhov — CHANGES_REQUESTED** [review]
LGTM

**Review by grok-rs — APPROVED** [review]
LGTM

**Pull request merged**